### PR TITLE
Developer branch (all feature branches merged)

### PR DESCRIPTION
**Pull request by John-Kibirige**
## Activities worked on 
- setup project with webpack
- setup linters (CSS, HTML, and Javascript)
- Use git flow to work on separate features with partner
- Made the website dynamic
- Used ES6 syntax 
- used callbacks and promises 
- used jest to do unit testing 
- solve conflicts
- used API documentation
- solved linter errors
- deployed website using github pages

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,86 @@
...
+- Technologies used: Linters, Git and Github, webpack and jest
+
+## Live Demo
+
+[https://john-kibirige.github.io/tv-maze/](https://john-kibirige.github.io/tv-maze/)
+
+## Video description
+
+[Video description](https://drive.google.com/file/d/1I-L7Mm9XUsEtWLLenc9ZfN78tgYzMHJ7/view?usp=sharing)
+
```
Comment (AmaduKamara) [inline README.md]: You guys did well on this project but there are a few details you skipped in the video presentation. Kindly take note of the points listed below about the video presentation:

- [x] The student recorded a video presenting the features of the project. ( Please talk about the key features of the project )
- [x] The maximum length of the video is 5 minutes. (Your video lasted for 5.11mins which is above the max time for the video. Kindly do your video within 5 minutes or 5 max 👍)
- [x] The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear. ( I will suggest you consider this point as the first point. You introduce yourselves, kind of like, "This is our JavaScript capstone project that required us to build an app that interacts with an external API and we used the tv-maze  API. We worked on these features: list the features of the project 👍")
- [x] All the members in the video should appear in the video and talk about the project.
- [ ] The video contains a highlight about a piece of the code the student finds interesting or is proud of making. ( At least one member should open a section of their code that he is proud of implementing. Don't explain everything about the code. You can show the code and briefly talk about what that code does in the app 👍)

**Kindly redo the video and follow the guides above in presenting your project**

File: tests/commentcounter.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+getc.getComment = jest.fn(() => Promise.resolve([
+  { item_id: 1, username: 'meshu', comment: 'Love it seriously' },
+  { item_id: 1, username: 'meshu', comment: 'Love it seriously' },
+]));
+
+describe('Test counter function', () => {
+  it('Test counter at length 2', async () => {
+    const result = await counter(1);
+    expect(result).toBe(2);
+  });
```
Comment (AmaduKamara) [inline tests/commentcounter.test.js:9]: - [x] The test is failing on this section. Kindly work on the test here so it passes. The screenshot below shows details of the error:

image

File: src/unit-tests/involvmentapi.test.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      expect(res).toBe('Created');
+    });
+  });
+
+  test('Getting the number of likes from the api ', () => {
+    getLikes().then((response) => {
+      expect(response.likes).toBe(5);
+    });
+  });
+});
```
Comment (AmaduKamara) [inline src/unit-tests/involvmentapi.test.js:17]: - [ ] Kindly work on the test on this codebase. Currently, you're having the following failing tests:

image

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly
✅ No linter error
✅ Good commit messages
✅ Good descriptive PR
✅ App is designed and works fine 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: tests/addcomment.test.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  it('Number of Fetch call 1', async () => {
+    await addComment(1);
+    expect(fetch).not.toBeCalledTimes(3);
+  });
+
+  it('Number of Fetch call', async () => {
+    await addComment(1);
+    expect(fetch).toBeCalledTimes(3);
+  });
+});
```
Comment (elmar8287) [inline tests/addcomment.test.js]: - [ ] Please, consider that this test did not pass:
Screen Shot 2022-09-03 at 17 11 03

File: README.md [hunk 5]
```diff
@@ -1 +1,86 @@
...
+### Deployment
+
+## Authors
+
+👤 **Melashu Amari**
+
+- [Github](https://github.com/melashu)
+- [Twitter](https://twitter.com/meshu102)
+- [LinkedIn](https://twitter.com/home)
```
Comment (elmar8287) [inline README.md:60]: - [ ] OPTIONAL Please, consider removing not related links to each author. Here is the LinkedIn link is linked to the home page 👍 You can remove it or fix the correct link 🥇

File: README.md [hunk 6]
```diff
@@ -1 +1,86 @@
...
+
+👤 **Kibirige John**
+
+- [Github](https://github.com/John-Kibirige)
+- [Twitter](https://twitter.com/kibirigejohn005)
+- [LinkedIn](https://www.linkedin.com/in/kibirige-john-64160520a/trk=public_profile_samename-profile_profile-result-card_result-card_full-click&original_referer=https%3A%2F%2Fwww%2Egoogle%2Ecom%2F&originalSubdomain=ug)
```
Comment (elmar8287) [inline README.md:66]: - [ ] OPTIONAL Please, consider removing not related links to each author. Here is the LinkedIn link linked to the home page 👍 You can remove it or fix the correct link 🥇

**Review by elmar8287 — CHANGES_REQUESTED** [review]
Hi, team.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights

- You have fixed the issues related to these tests: `involvmentapi.test.js` and `commentcounter.test.js`
 
## Required Changes ♻️

- [ ] Check the comments under the review.
- [ ] Please, consider fixing the sharing access for the video - actually I have no access to watch it. You have to share this video with everybody, who has the link 👍 



## Optional suggestions

- [ ] Actually, you did not consider the Git Flow: the default branch should be `dev`, not the `main` branch 👍 
Screen Shot 2022-09-03 at 17 02 05


- [ ] I have made some OPTIONAL suggestions under the review 👍 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
giphy

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — APPROVED** [review]
Hi @TEAM, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-  [ ] Kindly also include the optional suggestion into the project, this will make your project better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**